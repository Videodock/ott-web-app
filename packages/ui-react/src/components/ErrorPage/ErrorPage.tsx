import React, { type ReactNode } from 'react';
import { useTranslation } from 'react-i18next';
import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
import { IS_DEMO_MODE, IS_DEVELOPMENT_BUILD, IS_PREVIEW_MODE } from '@jwp/ott-common/src/utils/common';

import DevStackTrace from '../DevStackTrace/DevStackTrace';

import styles from './ErrorPage.module.scss';

interface Props {
  disableFallbackTranslation?: boolean;
  title?: string | ReactNode;
  message?: string | ReactNode;
  learnMoreLabel?: string;
  children?: React.ReactNode;
  error?: Error;
  helpLink?: string;
}

const ErrorPage = ({ title, message, learnMoreLabel, ...rest }: Props) => {
  const { t } = useTranslation('error');

  return (
    <ErrorPageWithoutTranslation
      title={title || t('generic_error_heading')}
      message={message || t('generic_error_description')}
      learnMoreLabel={learnMoreLabel || t('learn_more')}
      {...rest}
    />
  );
};

export const ErrorPageWithoutTranslation = ({ title, children, message, learnMoreLabel, error, helpLink }: Props) => {
  const logo = useConfigStore((s) => s.config?.assets?.banner);
  const alt = ''; // intentionally empty for a11y, because adjacent text alternative

  return (
    <div className={styles.errorPage}>
      <div className={styles.box}>
        <img className={styles.image} src={logo || '/images/logo.png'} alt={alt} />
        <h1 className={styles.title}>{title || 'An error occurred'}</h1>
        <div className={styles.main}>
          <p className={styles.message}>{message || 'Try refreshing this page or come back later.'}</p>
          {children}
          {(IS_DEVELOPMENT_BUILD || IS_DEMO_MODE || IS_PREVIEW_MODE) && helpLink && (
            <div className={styles.links}>
              <a href={helpLink} target={'_blank'} rel={'noreferrer'}>
                {learnMoreLabel || 'Learn More'}
              </a>
              {(IS_DEVELOPMENT_BUILD || IS_PREVIEW_MODE) && error?.stack && (
                <span className={styles.stack}>
                  <DevStackTrace error={error} />
                </span>
              )}
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

export default ErrorPage;
