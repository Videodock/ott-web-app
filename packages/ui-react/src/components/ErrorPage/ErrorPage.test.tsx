import React from 'react';
import { axe } from 'vitest-axe';
import { render } from '@testing-library/react';

import ErrorPage from './ErrorPage';

describe('<ErrorPage>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ErrorPage title="This is the title">This is the content</ErrorPage>);

    expect(container).toMatchSnapshot();
  });

  test('WCAG 2.2 (AA) compliant', async () => {
    const { container } = render(<ErrorPage title="This is the title">This is the content</ErrorPage>);

    expect(await axe(container, { runOnly: ['wcag21a', 'wcag21aa', 'wcag22aa'] })).toHaveNoViolations();
  });
});
