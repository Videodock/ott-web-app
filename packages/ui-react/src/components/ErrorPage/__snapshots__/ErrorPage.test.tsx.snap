// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ErrorPage> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_errorPage_8c5621"
  >
    <div
      class="_box_8c5621"
    >
      <img
        alt=""
        class="_image_8c5621"
        src="/images/logo.png"
      />
      <h1
        class="_title_8c5621"
      >
        This is the title
      </h1>
      <div
        class="_main_8c5621"
      >
        <p
          class="_message_8c5621"
        >
          generic_error_description
        </p>
        This is the content
      </div>
    </div>
  </div>
</div>
`;
